src/laws/laws_plots.jl

```diff
@@ -50,13 +50,24 @@ function plot_law_1d(
     inputs::NamedTuple,
     glacier_idx::Integer,
     θ,
-    input_names::Vector{Symbol}
+    input_name::Symbol
 )
-    xname = input_names[1]
-    xvals = get_input(inputs[xname], simulation, glacier_idx, 2010.0)
-    input_tuple = NamedTuple{(xname,)}((xvals,))
-    outputs = eval_law(law, simulation, glacier_idx, input_tuple, θ)
-    return plot(xvals, outputs, xlabel=string(xname), ylabel=string(law.name), title="Law Function Plot")
+    scalar = length(get_input(inputs[input_name], simulation, 1, 2010.0)) == 1 ? true : false
+    @show scalar
+    if scalar 
+        xvals = [get_input(inputs[input_name], simulation, i, 2010.0) for i in 1:length(simulation.glaciers)]
+        input_tuples = [NamedTuple{(input_name,)}((xvals[i],) ) for i in 1:length(simulation.glaciers)]
+        outputs = [only(eval_law(law, simulation, i, input_tuples[i], θ)) for i in 1:length(simulation.glaciers)]
```

Comment:
Why are you looping over the `glaciers` and not for the gridded case?
Wouldn't the code below make more sense?
```suggestion
        xvals = [get_input(inputs[input_name], simulation, glacier_idx, 2010.0)]
        input_tuples = [NamedTuple{(input_name,)}((xvals[1],) )]
        outputs = [only(eval_law(law, simulation, glacier_idx, input_tuples[1], θ))]
```